### PR TITLE
PP-12356 Implement POST /v1/api/service/{serviceId}/[test|live]/worldpay/check-credentials 

### PR DESCRIPTION
**Pull request by DomBelcher**
## WHAT YOU DID
 - Add `POST /v1/api/service/{serviceId}/[test|live]/worldpay/check-credentials` endpoint with tests
 - Enforce consistency for test names in `GatewayAccountCredentialsResourceTest` and `GatewayAccountCredentialsResourceIT`

## How to test
Run tests

### CODE REVIEW

File: src/test/java/uk/gov/pay/connector/gatewayaccountcredentials/resource/GatewayAccountCredentialsResourceIT.java [hunk 1]
```diff
@@ -58,9 +73,8 @@ void setUp() {
     
     @Nested
     class CreateGatewayAccountCredentials {
-
         @Test
-        void withCredentials_responseShouldBe200_Ok() {
+        void forValidCredentials_shouldReturn200_andCreateCredentials() {
```
Comment (oswaldquek) [inline src/test/java/uk/gov/pay/connector/gatewayaccountcredentials/resource/GatewayAccountCredentialsResourceIT.java:77]: I think it makes more sense to flip the test name here, for eg. `shouldReturn200_andCreateCredentials_forValidCreds`? And similarly for the rest of the test names in this PR.

File: src/test/java/uk/gov/pay/connector/gatewayaccountcredentials/resource/GatewayAccountCredentialsResourceIT.java [hunk 2]
```diff
@@ -298,6 +312,63 @@ void withInvalidCredentials_shouldReturn400() {
...
+                    .body("message[0]", is(format("Gateway account not found for service ID [%s] and account type [%s]", SERVICE_ID, TEST)));
+        }
+
+        @Test
+        void forValidCredentials_shouldReturn200_withResultValid() {
+            app.getWorldpayMockClient().mockCredentialsValidationValid();
+            
+            String gatewayAccountId = app.givenSetup()
+                    .body(toJson(Map.of(
+                            "payment_provider", "stripe",
```
Comment (oswaldquek) [inline src/test/java/uk/gov/pay/connector/gatewayaccountcredentials/resource/GatewayAccountCredentialsResourceIT.java:333]: Given we're testing the URL ending in `/worldpay/check-credentials`, it seems like a code smell that the test works for a gateway account with a Stripe PSP...

Comment (DomBelcher) [reply src/test/java/uk/gov/pay/connector/gatewayaccountcredentials/resource/GatewayAccountCredentialsResourceIT.java:333]: I'll admit that's down to me copying and pasting without thinking 🤦 interesting that it works though

Comment (oswaldquek) [reply src/test/java/uk/gov/pay/connector/gatewayaccountcredentials/resource/GatewayAccountCredentialsResourceIT.java:333]: Yeah it's probably a bug and not something you're responsible for. Be good to understand what's going on though.

File: src/test/java/uk/gov/pay/connector/gatewayaccountcredentials/resource/GatewayAccountCredentialsResourceIT.java (hunk 1 above)
Comment (DomBelcher) [reply src/test/java/uk/gov/pay/connector/gatewayaccountcredentials/resource/GatewayAccountCredentialsResourceIT.java:77]: In #5228 I went for `for...should` - my thinking was it was analogous to the way the test is written, i.e. you set up the test, run the code under test and then assert on the result. 

I do see the argument for `should...for` though, and don't have any strong attachment one way or the other. I'm happy with picking a standard and then enforcing it across the board - I don't mind going back and changing existing tests to `should...for`

Comment (oswaldquek) [reply src/test/java/uk/gov/pay/connector/gatewayaccountcredentials/resource/GatewayAccountCredentialsResourceIT.java:77]: Yeah I think we should write test/method names in natural English as much as possible. It's more natural to say "should return 200 and create creds xxx for valid creds".

**Comment by DomBelcher** [comment]
Closing as this is implemented in #5282

**Pull request closed without merging**